core/src/main/java/com/linecorp/armeria/common/stream/StreamMessage.java

```diff
@@ -527,4 +527,23 @@ default EventExecutor defaultSubscriberExecutor() {
 
         return FuseableStreamMessage.of(this, function);
     }
+
+    /**
+     * Resumes a failed {@link StreamMessage} by subscribing to a returned fallback {@link StreamMessage}
+     * when any error occurs.
+     *
+     * <p>Example:<pre>{@code
+     * DefaultStreamMessage<Integer> stream = new DefaultStreamMessage<>();
+     * stream.write(1);
+     * stream.write(2);
+     * stream.close(new IllegalStateException("Oops...");
+     * StreamMessage<Integer> resumed = stream.resume(cause -> StreamMessage.of(3, 4));
+     *
+     * assert resumed.collect().join().equals(List.of(1, 2, 3, 4));
+     * }</pre>
+     */
+    default StreamMessage<T> resume(Function<? super Throwable, ? extends StreamMessage<T>> function) {
```

Comment:
`resume` doesn't convey its behavior that recovers from an exception. How about renaming to `recoverAndResume()`?